snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -153,8 +155,9 @@ public boolean databaseExists(SnowflakeIdentifier database) {
                   queryHarness.query(
                       conn, finalQuery, SCHEMA_RESULT_SET_HANDLER, database.databaseName()));
     } catch (SQLException e) {
-      if (e.getErrorCode() == 2003 && e.getMessage().contains("does not exist")) {
-        return false;
+      // 2003, 2043, 2001 are possible Snowflake error codes for Object does not exist
+      if (e.getErrorCode() == 2003 || e.getErrorCode() == 2043 || e.getErrorCode() == 2001) {
```

Comment:
One suggestion would be to create sets for each kind of error (SCHEMA_NOT_FOUND or DATABASE_NOT_FOUND) etc. which could be reused in same context. This would help in future cases where the error diverged for a certain scenario like database.

Reply:
Yeah, I think the main thing is to assign intention/expectations to each set for readability. Especially in cases of errors originating from integer codes, it's better to err on the side of being more verbose.

Sets for each kind of error seem a reasonable starting point.